exporter/src/main/java/io/zeebe/exporter/spi/Exporter.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.zeebe.exporter.spi;
+
+import io.zeebe.exporter.context.Configuration;
+import io.zeebe.exporter.context.Context;
+import io.zeebe.exporter.context.Controller;
+import io.zeebe.exporter.record.Record;
+
+/**
+ * Minimal interface to be implemented by concrete exporters.
+ *
+ * <p>A concrete implementation should always provide a default, no-arguments constructor. It is not
+ * recommended to do anything in the constructor, but rather use the provided callbacks for
+ * configuration, setup, resource allocation, etc.
+ */
+public interface Exporter {
+  /**
+   * Use the provided configuration at this point to configure your exporter. This method is called
+   * right before opening the exporter.
+   *
+   * @param configuration the exporter configuration
+   */
+  default void configure(Configuration configuration) {}
+
+  /**
+   * Hook to perform any validation when the exporter is loaded. This method is called exactly once
+   * at startup when the exported is loaded, allowing an exporter to fail early if mis-configured.
+   * It's not recommended to perform complex or slow validation here, as it will block the startup
+   * of the broker.
+   *
+   * <p>If an exception is thrown, or the method returns false, the broker will not start.
+   *
+   * @param configuration parsed configuration
+   * @return true if valid, false otherwise
+   * @throws Exception any validation exception
+   */
+  default boolean validate(Configuration configuration) throws Exception {
```

Comment:
Why do I need to verify the configuration in an extra method?

Reply:
The idea of this method is that it is called once at startup to provide a fail-fast method and prevent the broker from starting should the exporter be misconfigured. I thought about conflating that with `configure` but I thought having an explicit method was better. I'm not opposed to going back to having both and documenting the two use cases.